delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ResourcesRoutes.scala

```diff
@@ -248,4 +263,30 @@ object ResourcesRoutes {
       fusionConfig: FusionConfig
   ): Route = new ResourcesRoutes(identities, aclCheck, resources, projectsDirectives, index).routes
 
+  implicit private val api: JsonLdApi = JsonLdJavaApi.lenient
+
+  def asSourceWithMetadata(
+      resource: ResourceF[Resource]
+  )(implicit baseUri: BaseUri, cr: RemoteContextResolution): IO[ResourceRejection, Json] = {
+    metadataJson(resource)
+      .map(mergeOriginalPayloadWithMetadata(resource.value.source, _))
+  }
+
+  private def metadataJson(resource: ResourceF[Resource])(implicit baseUri: BaseUri, cr: RemoteContextResolution) = {
+    implicit val resourceFJsonLdEncoder: JsonLdEncoder[ResourceF[Unit]] = ResourceF.defaultResourceFAJsonLdEncoder
+    resourceFJsonLdEncoder
+      .compact(resource.void)
+      .map(_.json)
+      .mapError(e => InvalidJsonLdFormat(Some(resource.id), e))
+  }
+
+  private val IdLens = root.`@id`.string
```

Comment:
We haven't really used `circe-optics` outside of migration code and tests so far.
It has not been really well maintained lately, so it would be better to stick to vanilla circe.